### ISSUE TITLE
Unverified session gate not working on password change old settings page

### ISSUE THREAD

**Issue by 6a68** [mozilla/fxa#7288]
If you change your password from an unverified session, you'll be able to enter the password, but not save it, and no error is displayed.

The easiest way I've found to trigger this is to change password, then try to change it again.

Copied over from https://bugzilla.mozilla.org/show_bug.cgi?id=1686793, filed 14 January.

**Comment by chenba** [comment]
@6a68 did you have 2fa enabled when you reproduced this bug?

**Comment by vasilicatamas** [comment]
This issue reproduces only with 2fa enabled and it is an old issue tracked in 4257 and 4753

**Comment by chenba** [comment]
I suspect the original bug as reported in Bugzilla is a different one as it didn't require a password change to get the session to a unverified state.

**Comment by chenba** [comment]
@vasilicatamas are you able to reproduce the bug using the steps described in the bugzilla bug?

**Comment by vasilicatamas** [comment]
I was able to reproduce the error following the same steps as 6a68 mentioned in the first comment, by changing the password two consecutive times: first time in the old settings page and the second one in the new settings page or 2 consecutive times in the new settings page.

2FA must be enabled in order to reproduce this issue.

![Screenshot_40](https://user-images.githubusercontent.com/18114210/105360335-24f9b780-5c01-11eb-9555-8f7084532ed4.png)

**Comment by chenba** [comment]
> I was able to reproduce the error following the same steps as 6a68 mentioned in the first comment

Yes, I can as well.  The original Bugzilla bug has the following STR:

> Click menu/email address/Manage Account/Password/Change Password

That sounds like the user is coming into settings from the browser sync menu.  And he saw the bug on the _first_ password change attempt.

I couldn't reproduce with those steps.
